src/panels/lovelace/common/validate-condition.ts

```diff
@@ -308,3 +312,43 @@ export function addEntityToCondition(
   }
   return condition;
 }
+
+export function extractMediaQueries(conditions: Condition[]): string[] {
+  return conditions.reduce<string[]>((array, c) => {
+    if ("conditions" in c && c.conditions) {
+      array.push(...extractMediaQueries(c.conditions));
+    }
+    if (c.condition === "screen" && c.media_query) {
+      array.push(c.media_query);
+    }
+    return array;
+  }, []);
+}
+
+export function createConditionMediaQueriesListeners(
+  conditions: Condition[],
+  hass: HomeAssistant,
+  onChange: (visibility: boolean) => void
+): MediaQueriesListener[] {
+  const mediaQueries = extractMediaQueries(conditions);
+
+  const listeners: MediaQueriesListener[] = [];
+  mediaQueries.forEach((query) => {
+    const listener = listenMediaQuery(query, (matches) => {
+      // For performance, if there is only one condition and it's a screen condition, set the visibility directly
+      if (
```

Comment:
Instead of checking this inside the if-statement, you can pass a different function as 2nd arg to `listenMediaQuery`. Or maybe even move it above the forEach.